cns/restserver/internalapi.go

```diff
@@ -94,6 +94,15 @@ func (service *HTTPRestService) SyncNodeStatus(dncEP, infraVnet, nodeID string,
 	}
 	service.RUnlock()
 
+	ncVersionListResp, err := service.nma.GetNCVersionList(context.TODO())
```

Comment:
For all of these NMAgent methods, a `context.Context` is required to control the request. Please plumb in the incoming request to CNS (available via `http.(*Request).Context()` so that requests to NMAgent will be retried only as long as the upstream requestor to CNS is willing to wait (retries will occur indefinitely with a TODO).

There's several instances of `context.TODO()` in this patch... won't flag them to keep the noise down.

Reply:
Updated with timeout.